src/main/java/edu/cornell/kfs/tax/dataaccess/impl/TransactionRowPRNCBuilder.java

```diff
@@ -0,0 +1,544 @@
+                    }
+                }
+            }
+            
+            incomeCodeSubType = summary.incomeClassCodeToIrsIncomeCodeSubTypeMap.get(incomeClassCode);
+            if (StringUtils.isBlank(incomeCodeSubType)) {
+                // If no subtype was found, then set it to the exclusion value.
+                incomeCodeSubType = summary.excludedIncomeCodeSubType;
+                numExcludedAssignedIncomeCodeSubTypes++;
+            }
+            
+            // Prepare to update income code and income code subtype fields.
+            rs.updateString(summary.transactionDetailRow.incomeCode.index, incomeCode);
+            rs.updateString(summary.transactionDetailRow.incomeCodeSubType.index, incomeCodeSubType);
+        }
+        
+        @Override
+        EnumMap<TaxStatType,Integer> getStatistics() {
+            EnumMap<TaxStatType,Integer> stats = super.getStatistics();
+            
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODES, Integer.valueOf(numExcludedAssignedIncomeCodes));
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODES_DV, Integer.valueOf(numExcludedAssignedIncomeCodes));
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODE_SUBTYPES, Integer.valueOf(numExcludedAssignedIncomeCodeSubTypes));
+            stats.put(TaxStatType.NUM_EXCLUDED_ASSIGNED_INCOME_CODE_SUBTYPES_DV, Integer.valueOf(numExcludedAssignedIncomeCodeSubTypes));
+            stats.put(TaxStatType.NUM_DETERMINED_FED_TAX_WITHHELD_INCOME_CODES, Integer.valueOf(numDeterminedFederalTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_DETERMINED_FED_TAX_WITHHELD_INCOME_CODES_DV, Integer.valueOf(numDeterminedFederalTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_UNDETERMINED_FED_TAX_WITHHELD_INCOME_CODES, Integer.valueOf(numUndeterminedFederalTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_DETERMINED_STATE_INC_TAX_WITHHELD_INCOME_CODES, Integer.valueOf(numDeterminedStateIncomeTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_DETERMINED_STATE_INC_TAX_WITHHELD_INCOME_CODES_DV, Integer.valueOf(numDeterminedStateIncomeTaxWithheldIncomeCodes));
+            stats.put(TaxStatType.NUM_UNDETERMINED_STATE_INC_TAX_WITHHELD_INCOME_CODES, Integer.valueOf(numUndeterminedStateIncomeTaxWithheldIncomeCodes));
+            
+            return stats;
+        }
+        
+    }
+    
+    abstract TaxTableField getExtraField(TransactionDetailRow detailRow);
+
+
+
+    @Override
+    String getSqlForSelectingCreatedRows(T summary) {
+        return TaxSqlUtils.getTransactionDetailSelectSql(getExtraField(summary.transactionDetailRow), summary.transactionDetailRow, true, false);
+    }
+
+    @Override
+    Object[][] getParameterValuesForSelectingCreatedRows(T summary) {
+        return new Object[][] {
+            {summary.reportYear},
+            {CUTaxConstants.NEEDS_UPDATING_BOX_KEY}
+        };
+    }
```

Comment:
There appears to be some additional methods listed after the nested classes. Perhaps things would be clearer if they were moved to before the nested classes in this file.